include/grpcpp/impl/codegen/byte_buffer.h

```diff
@@ -96,6 +96,11 @@ class ByteBuffer final {
   /// Dump (read) the buffer contents into \a slices.
   Status Dump(std::vector<Slice>* slices) const;
 
+  /// Returns true is buffer only holds one slice.
```

Comment:
I feel like this does not qualify as an API. Can we add a friend and use `c_buffer()`?

Reply:
Agreed. Let this be a friend if it is just a fastpath for performance reasons between pieces of our own code. Avoids the gRFC issue as well that way.

Reply:
Done

Reply:
I had to add 
```
#include <grpcpp/impl/codegen/config_protobuf.h>
```

to byte_buffer.h, so I could add the friend declaration
```
  template <class ProtoBufferReader, class T>
  friend Status GenericDeserialize(ByteBuffer* buffer,
                                   grpc::protobuf::Message* msg);
```

Is that ok? or does it bring in proto deps? ByteBuffer should be usable w/o proto deps

Also, I tried to fwd decl it with:
```
namespace protobuf {
  class Message
}
```

but that caused all sorts of errors